src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -33,24 +34,32 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc(options =>
-                {
-                    options.Filters.Add(new ProducesAttribute("application/json"));
-
-                    options.EnableEndpointRouting = false;
-                })
-                .SetCompatibilityVersion(CompatibilityVersion.Latest)
-                .AddApplicationPart(typeof(DiagController).Assembly);
+            // AddControllers is sufficient because the tool does not use Razor nor Views.
+            services.AddControllers(options =>
+            {
+                options.EnableEndpointRouting = false;
+            })
+            .AddJsonOptions(options =>
+            {
+                // Allow serialization of enum values into strings rather than numbers.
+                options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+            })
+            .SetCompatibilityVersion(CompatibilityVersion.Latest)
+            .AddApplicationPart(typeof(DiagController).Assembly);
 
             services.Configure<ApiBehaviorOptions>(options =>
             {
                 options.InvalidModelStateResponseFactory = context =>
                 {
                     var details = new ValidationProblemDetails(context.ModelState);
                     var result = new BadRequestObjectResult(details);
-                    result.ContentTypes.Add("application/problem+json");
+                    result.ContentTypes.Add(ContentTypes.ApplicationProblemJson);
                     return result;
                 };
+
+                // Suppress the automatic mapping of 4XX status codes to the ProblemDetails type.
+                // The tool uses ValidationProblemDetails and explicitly handles exceptions with this type.
+                options.SuppressMapClientErrors = true;
```

Comment:
Have we looked into what this looks like if we do automatic mapping?

Reply:
Without suppressing this, the 401 responses have an additional content types of application/json and application/problem+json with a structure of ProblemDetails, which is also added to the schema. This occurs because we state that 401 does not have a response body (using typeof(void)) and API behavior middleware interprets that it should override that with the ProblemDetails type.

I can work around this by specifying the ProducesErrorResponseType attribute in stead of suppressing the mapping of client errors.